solc/CommandLineParser.cpp

```diff
@@ -0,0 +1,1243 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include "license.h"
+
+#include <solc/CommandLineParser.h>
+#include <libyul/optimiser/Suite.h>
+#include <liblangutil/EVMVersion.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <range/v3/view/transform.hpp>
+
+using namespace std;
+using namespace solidity::langutil;
+
+namespace po = boost::program_options;
+
+namespace solidity::frontend
+{
+
+static bool g_hasOutput = false;
+
+namespace
+{
+
+std::ostream& sout()
+{
+	g_hasOutput = true;
+	return cout;
+}
+
+std::ostream& serr(bool _used = true)
+{
+	if (_used)
+		g_hasOutput = true;
+	return cerr;
+}
+
+}
+
+#define cout
+#define cerr
+
+static string const g_strAbi = "abi";
+static string const g_strAllowPaths = "allow-paths";
```

Comment:
Another question, I think it's needless to duplicate all these strings out of CDATA into the heap. Why not using `string_view` (which we should embrace IIRC) since lately?

Reply:
Again, I agree but I'm going to change it in a separate PR.